src/esp/gfx/Simulator.h

```diff
@@ -263,6 +260,16 @@ class Simulator {
    */
   double getWorldTime();
 
+  /**
+   * @brief Compute the navmesh for the simulator's current active scene.
+   * @param navMeshSettings The @ref nav::NavMeshSettings instance to
+   * parameterize the navmesh construction.
+   * @return The resulting completed @ref Pathfinder object or nullptr if
+   * failed.
+   */
+  std::shared_ptr<nav::PathFinder> recomputeNavMesh(
+      nav::NavMeshSettings& navMeshSettings);
```

Comment:
This should be either `const&` or explicitly documented that it's a `@param[in,out]` and *why* a mutable access is needed.